pkg/awsutils/awsutils.go

```diff
@@ -827,11 +827,39 @@ func (cache *EC2InstanceMetadataCache) createENI(useCustomCfg bool, sg []*string
 		},
 	}
 
-	input := &ec2.CreateNetworkInterfaceInput{
-		Description:       aws.String(eniDescription),
-		Groups:            aws.StringSlice(cache.securityGroups.SortedList()),
-		SubnetId:          aws.String(cache.subnetID),
-		TagSpecifications: tagSpec,
+	var needIPs = numIPs
+
+	ipLimit := cache.GetENIIPv4Limit()
+
+	if ipLimit < needIPs {
+		needIPs = ipLimit
+	}
+
+	if needIPs < 1 {
+		return "", nil
+	}
```

Comment:
nit: let's add a comment here explaining that we cannot allocate more IPs for an ENI than it can handle